iree/turbine/kernel/wave/utils.py

```diff
@@ -219,18 +219,30 @@ def is_mma(node):
         rhs_shape = custom.rhs_type.symbolic_shape
         acc_shape = custom.acc_type.symbolic_shape
         k = ((set(lhs_shape) & set(rhs_shape)) - set(acc_shape)).pop()
-        mapping[m] = 0
-        mapping[n] = 1
-        mapping[k] = 2
-        # Update vector shapes in hardware constraint.
-        M, N, K = hardware_constraint.mma_matrix_shapes
-        if not hardware_constraint.vector_shapes:
-            hardware_constraint.vector_shapes = {}
-        hardware_constraint.vector_shapes[m] = M
-        hardware_constraint.vector_shapes[n] = N
-        hardware_constraint.vector_shapes[k] = K
-
-    return mapping, capture_mma_slices([get_custom(x) for x in mma_nodes])
+        if custom not in mapping:
+            mapping[custom] = {}
+        mapping[custom][m] = 0
+        mapping[custom][n] = 1
+        mapping[custom][k] = 2
```

Comment:
No actions for now: Would be nice in the future to have an enum to represent [0, 1, 2] as [M, N, K] for better legibiltiy

Reply:
makes sense, will add that as well.